src/api/java/appeng/api/networking/IGridBlock.java

```diff
@@ -68,37 +74,46 @@
 	boolean isWorldAccessible();
 
 	/**
+	 * Must not return when {@link #isWorldAccessible()} is true.
+	 * Otherwise the behavior is unspecified.
+	 *
 	 * @return current location of this node
 	 */
+	@Nonnull
 	DimensionalCoord getLocation();
 
 	/**
 	 * @return Transparent, or a valid color, NULL IS NOT A VALID RETURN
 	 */
+	@Nonnull
 	AEColor getGridColor();
 
 	/**
 	 * Notifies your IGridBlock that changes were made to your connections
 	 */
-	void onGridNotification( GridNotification notification );
+	void onGridNotification( @Nonnull GridNotification notification );
 
 	/**
 	 * Update Blocks network/connection/booting status. grid,
 	 *
 	 * @param grid grid
 	 * @param channelsInUse used channels
+	 * @deprecated to be removed in rv7
 	 */
+	@Deprecated
 	void setNetworkStatus( IGrid grid, int channelsInUse );
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1133)